examples/background.rb

```diff
@@ -0,0 +1,39 @@
+require 'optparse'
+$LOAD_PATH.unshift("../lib") # manageiq-messaging
+
+require 'manageiq-messaging'
+require_relative "common"
+
+Thread::abort_on_exception = true
+
+class ProducerConsumer < Common
+  SERVICE_NAME = "prodcom"
+  def run
+    client_loop(q_options) do |client|
+      puts "producer"
+      5.times do |i|
+        client.publish_message(:service    => SERVICE_NAME,
+                               :class_name => self.class.name,
+                               :message    => 'do_stuff',
+                               :payload    => {:args=>["hello#{i}"]}
+                              )
+      end
+      puts "produced"
+
+      # this could be a different client
+      puts "consumer"
+      client.subscribe_background_job(:service => SERVICE_NAME)
+
+      sleep(10) # wait for consumer to consume all
+      puts "consumed"
+    end
+  end
+
+  def self.do_stuff(args)
+    puts "GOT MESSAGE: do_stuff(#{args.inspect})"
+  end
+end
+
+if __FILE__ == $0
```

Comment:
If this file is itself an executable (rather than a file you would require), then you don't need to add the `__FILE__ == $0` bit (i.e. when will you ever require this file without executing it)